File: backend/src/main/java/com/devconnect/models/Comment.java
package com.devconnect.models;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Data
public class Comment {
    private UUID id = UUID.randomUUID();
    
    @NotBlank
    private String content;
    
    private List<CodeBlock> codeBlocks = new ArrayList<>();
    
    @NotBlank
    private UUID authorId;
    
    @NotBlank
    @Size(min = 3, max = 30)
    private String authorAlias;
    
    private boolean isAiGenerated;
    
    private UUID postId;
    
    private Instant createdAt = Instant.now();
}

File: android/app/src/main/kotlin/com/devconnect/data/model/Comment.kt
package com.devconnect.data.model

import java.time.Instant
import java.util.UUID

data class Comment(
    val id: UUID,
    val content: String?,
    val codeBlocks: List<CodeBlock>,
    val authorId: UUID,
    val authorAlias: String,
    val postId: UUID,
    val createdAt: Instant,
    val isAiGenerated: Boolean,
    val aiMetadata: AiMetadata?,
    val reactions: Map<ReactionType, Int>
) {
    enum class ReactionType {
        HELPFUL, INSIGHTFUL
    }
}

data class AiMetadata(
    val model: String,
    val promptContext: String
)
